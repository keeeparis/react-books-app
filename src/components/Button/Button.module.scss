.button {
  cursor: pointer;
  appearance: none;
  position: relative;
  border: none;
  padding: 10px 20px;
  font-family: inherit;
  font-size: 16px;
  border-radius: 8px;
  background: var(--color-3-dark);
  outline-color: var(--color-1);
  color: white;
  transition: all 0.2s ease;
  box-shadow: 1px 6px 20px 1px rgb(0 0 0 / 14%);

  &:hover:not(:disabled) {
    transition: all 0.2s ease;
    background: var(--color-3);
  }

  &:disabled {
    cursor: not-allowed;
  }

  &:focus-visible {
    outline-width: 3px;
  }

  &:active:not(:disabled) {
    transition: all 0.2s ease;
    transform: translateX(10px);
    outline-width: 1px;
  }
}
