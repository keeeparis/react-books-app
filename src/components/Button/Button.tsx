import React, { FC, HtmlHTMLAttributes } from 'react'
import styles from './Button.module.scss'

interface ButtonProps extends HtmlHTMLAttributes<HTMLButtonElement> {
  type?: 'submit' | 'button' | 'reset'
  disabled?: boolean
}

const Button: FC<ButtonProps> = ({ children, ...props }) => {
  return (
    <button {...props} className={styles.button}>
      {children}
    </button>
  )
}

export default Button
