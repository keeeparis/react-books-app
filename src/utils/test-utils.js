import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
