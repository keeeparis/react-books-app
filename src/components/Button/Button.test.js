import { render, screen, fireEvent, userEvent } from 'test-utils'
import Button from './Button'

it('test Button Component', () => {
  const { asFragment } = render(<Button />)

  expect(asFragment()).toMatchSnapshot()
})
