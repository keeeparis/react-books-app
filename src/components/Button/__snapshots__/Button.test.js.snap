// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button Component 1`] = `
<DocumentFragment>
  <button
    class="button"
  />
</DocumentFragment>
`;
